tool/src/org/antlr/v4/codegen/Target.java

```diff
@@ -504,9 +506,10 @@ public boolean grammarSymbolCausesIssueInGeneratedCode(GrammarAST idNode) {
 
 	public boolean templatesExist() {
 		String groupFileName = CodeGenerator.TEMPLATE_ROOT + "/" + getLanguage() + "/" + getLanguage() + STGroup.GROUP_FILE_EXTENSION;
+		URL url = Thread.currentThread().getContextClassLoader().getResource(groupFileName);
 		STGroup result = null;
 		try {
-			result = new STGroupFile(groupFileName);
+			result = new STGroupFile(url);
```

Comment:
It looks to me as though STGroupFile's ctor calls `getURL()` which does exactly that getResource thing. What additional functionality are you providing? i.e., where does ST4 go wrong? Maybe we shoulda fixed that in ST4 :)

Reply:
I believe is related to when it is called. I’d have to dig into it again.

Reply:
It's also possible the PR you merged into ST4 instead of mine (199) makes this unnecessary.

Reply:
ah. cool. if you have time, could you find the minimal changeset that works? I'm terrified of changing this stuff in the tool!

Reply:
Tested without the classloader in ANTLR (instead letting ST4 + getURL do the context class loader lookup) and it works. PR updated to what I believe is a minimal changeset.